### DISCUSSION TITLE
Has anyone noticed the reports not outputting Constraint Report?

### DISCUSSION THREAD

**Discussion by COF-RNorman** [ormsolutions/NORMA#46] [Q&A] [answered]
So question, has anyone noticed that with a new model you are not getting the Constraint report output? Not sure if this is a new issue or not so wanted to ask here first. Some of my older ORM files are generating the report fine. So it might be a file issue but not sure if it is just me.

![image](https://github.com/ormsolutions/NORMA/assets/73909402/dc37a1f5-ea96-4c32-8f45-ea16e7f896a0)

I think there might be a bad format or bug in the report generation but wanted to check before creating the issue. The rest of the application seems to be operating as expected so I don't think the file itself is bad. Just that it is not generating. I even simplified down to a single Object, 2 value types and 2 binary roles. This is also failing to generate a report at all.

![image](https://github.com/ormsolutions/NORMA/assets/73909402/f29a9d1f-083f-4eca-9021-d233aacbf44c)

**Comment by mcurland** [comment, marked as answer]
Wow, based on the code this has never worked due to the simplicity of the model. The generator uses either a filtered list or the 'all' list of object types and fact types. The failure is trying to sort the 'all' list, which isn't editable without copying the contents out. You don't need anything special for a filtered model: any spanning uniqueness constraint, explicit objectification or unary fact type will put you in a filtered state. The vast majority of models have at least one of these elements.

This is fixed in NORMA 1.0.3049.1. If you have automatic updates on you'll get the updated version by restarting Visual Studio.

Thanks for the report.

**Reply by COF-RNorman** [reply]
Awesome... Thanks for the quick check. I thought I was losing my mind or did something wrong to my files. LOL.

Make more sense now that you explained.

**Reply by COF-RNorman** [reply]
Confirmed that everything generated. Thanks again.

![image](https://github.com/ormsolutions/NORMA/assets/73909402/a5f7b867-d5b8-4753-88b8-f3bee67c73df)

**Comment by mcurland** [comment]
Don't worry, I thought you were losing your mind as well when I first read the report :)

Then I thought I'd broken something adding the Word Report in Pro, but that didn't change anything anywhere near this code.

Anyway, easy fix. I actually deleted a few lines or code.
